db/migrate/20221202130039_fill_project_journals_with_existing_data.rb

```diff
@@ -0,0 +1,95 @@
+class FillProjectJournalsWithExistingData < ActiveRecord::Migration[7.0]
+  def up
+    create_journal_entries_for_projects
+  end
+
+  def down
+    delete_journal_entries_for_projects
+  end
+
+  def create_journal_entries_for_projects
+    sql = <<~SQL.squish
+      WITH project_journals_insertion AS (
+        INSERT INTO project_journals(
+            name,
+            description,
+            public,
+            parent_id,
+            identifier,
+            active,
+            templated
+          )
+        SELECT name,
+          description,
+          public,
+          parent_id,
+          identifier,
+          active,
+          templated
+        FROM projects
+        RETURNING id,
+          identifier
+      ),
+      journals_insertion AS (
+        INSERT into journals (
+            journable_id,
+            journable_type,
+            user_id,
+            created_at,
+            version,
+            activity_type,
+            data_type,
+            data_id
+          )
+        SELECT projects.id,
+          'Project',
+          :user_id,
+          projects.created_at,
+          1,
+          'project_attributes',
+          'Journal::ProjectJournal',
+          project_journals_insertion.id
+        FROM projects
+          LEFT JOIN project_journals_insertion ON projects.identifier = project_journals_insertion.identifier
```

Comment:
The result should be equal but for the paranoid developer in me, I'd prefer a FULL JOIN over a LEFT JOIN.

Reply:
1b2e668f974f6702e5958fb9b2f7188008e923a2